cmd/thv-operator/pkg/sources/storage_manager_test.go

```diff
@@ -429,3 +430,230 @@ func TestConfigMapStorageManager_GetStorageReference(t *testing.T) {
 	assert.NotNil(t, ref.ConfigMapRef)
 	assert.Equal(t, "test-registry-registry-storage", ref.ConfigMapRef.Name)
 }
+
+func TestConfigMapStorageManager_ConfigureDeployment(t *testing.T) {
+	t.Parallel()
+
+	scheme := runtime.NewScheme()
+	require.NoError(t, mcpv1alpha1.AddToScheme(scheme))
+	require.NoError(t, corev1.AddToScheme(scheme))
+
+	tests := []struct {
+		name                  string
+		registry              *mcpv1alpha1.MCPRegistry
+		deployment            *appsv1.Deployment
+		containerName         string
+		expectError           bool
+		errorContains         string
+		expectedArgs          []string
+		expectedVolumeName    string
+		expectedVolumePath    string
+		expectedConfigMapName string
+	}{
+		{
+			name: "configure deployment successfully",
+			registry: &mcpv1alpha1.MCPRegistry{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-registry",
+					Namespace: "test-namespace",
+				},
+			},
+			deployment: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name: "registry-api",
+									Args: []string{"serve"},
+								},
+							},
+						},
+					},
+				},
+			},
+			containerName:         "registry-api",
+			expectError:           false,
+			expectedArgs:          []string{"serve", "--file=/data/registry/registry.json"},
```

Comment:
this `/data/registry/registry.json` should be a new constant or a function using the 2 constants, the mount path and ConfigMapStorageDataKey